hipogram/posts/views.py

```diff
@@ -38,63 +38,37 @@ def get_context_data(self):
         return context
 
 
-"""
-class ShareView(CreateView):
+class PostCreateView(CreateView):
     model = Post
-    fields = ['image', 'text', 'created_by', 'tags', 'creation_datetime']
+    fields = ['image', 'text', 'tags']
     template_name = "share.html"
-    form = 'PostForm'
-"""
+    success_url = reverse_lazy("posts:list")
+    pk_url_kwarg = 'post_id'
 
-
-@login_required()
-def post_new(request):
-    if request.method == "POST":
-        form = PostForm(request.POST, request.FILES)
+    # override this method to determine the creator of the post
+    def post(self, request, *args, **kwargs):
+        form = self.get_form()
         if form.is_valid():
             form.instance.created_by = request.user
-            form.save()
-            return redirect("posts:list")
-    else:
-        form = PostForm()
-    return render(request, 'share.html', {'form': form})
-
-
-def delete_post(request, post_id):
-    post = get_object_or_404(Post, id=post_id)
-
-    if request.user == post.created_by:
-        if request.method == 'POST':
-            form = PostForm(request.POST, instance=post)
-            post.delete()
-            messages.success(request, 'You have successfully deleted the post')
-            return redirect("posts:list")
+            return self.form_valid(form)
         else:
-            form = PostForm(instance=post)
-
-        return render(request, "delete.html", {'form': form})
-    else:
-        messages.error(request, 'You are not the owner of this post, please log in.')
-        return redirect("users:login")
+            return self.form_invalid(form)
```

Comment:
Does this work? It seems it's not okay.

Reply:
It works, but it could be better.

Reply:
Alright, you can continue with other views (like & rate), success / fail messages, login / owner required implementations. We can try to make this better in the meeting.